File: solutions/chapter3/problem_3_6.cpp
#include <iostream>
using namespace std;

int main() {
    int K, N;
    cin >> K >> N;

    int ans = 0;
    for (int x = 0; x <= K; x++) {
        for (int y = 0; y <= K; y++) {
            if (0 <= N - x - y && N - x - y <= K) ans++;
        }
    }
    cout << ans << endl;
}

File: solutions/chapter3/problem_3_6.py
def main():
    K, N = map(int, input().split())

    ans = 0
    for x in range(K+1):
        for y in range(K+1):
            if 0 <= N - x - y and N - x - y <= K:
                ans += 1

    print(ans)
    return

if __name__ == '__main__':
    main()